.github/workflows/deploy-server.yml

```diff
@@ -0,0 +1,69 @@
+name: Deploy to Amazon ECS
+
+on:
+  pull_request: # for now, trigger this on PR so I can test it before merging
+
+env:
+  AWS_REGION: us-east-2 # set this to your preferred AWS region, e.g. us-west-1
+  ECR_REPOSITORY: cribbage # set this to your Amazon ECR repository name
+  ECS_SERVICE: nginx-service # set this to your Amazon ECS service name
+  ECS_CLUSTER: cribbage # set this to your Amazon ECS cluster name
+  ECS_TASK_DEFINITION:
+    ./ecs-task-definition.json # set this to the path to your Amazon ECS task definition
+    # file, e.g. .aws/task-definition.json
+  CONTAINER_NAME:
+    nginx # set this to the name of the container in the
+    # containerDefinitions section of your task definition
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          # Build a docker container and
+          # push it to ECR so that it can
+          # be deployed to ECS.
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
```

Comment:
🔍 We have a workflow that builds and pushes out to [docker hub](https://hub.docker.com/r/joshprzybyszewski/cribbage), which is what I've been using. We may be able to avoid pushing to an ECR repo...

Reply:
Yes, we can consume the image from there if you want. Check out the image definition in the stack:
https://github.com/joshprzybyszewski/cribbage/blob/2a7c1572acf375429859ad0e98d6d64b001150ff/infrastructure/lib/infrastructure-stack.ts#L14-L16

As it is written, this will build the docker image based on the dockerfile in the repo if changes have occurred; we can also use `ecs.ContainerImage.fromRegistry` to pull from dockerhub; what are some benefits to staying away from ECR?